### PR TITLE
Added file sum of triplet in C

### PR DESCRIPTION
**Pull request by akanksha-saini**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?)

Yes.

### Description

Program in C to print all triplets of numbers which sum to target.
For example
input:
9
5 7 9 1 2 4 6 8 3
10
output:
1, 2 and 7
1, 3 and 6
1, 4 and 5
2, 3 and 5

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#585

### CODE REVIEW

File: C/cp/sum of triplet.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/*Take as input N, the size of array. Take N more inputs and store that in an array. Take as input �target�, a number. Write a function which prints all triplets of numbers which sum to target.
```
Comment (iamrajiv) [inline C/cp/sum of triplet.c:1]: @akanksha-saini add comments after this `/*`

File: C/cp/sum of triplet.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	
+	scanf("%d",&target); //input the target number
+	sort(a,n); // sort the array 
+    
+		
+	tripletSum(a,n,target); //cal tripletSum() to find and print triplet 
+	
+	return 0;
+}
+/* Time Complexity of sort function : O(n^2)
```
Comment (iamrajiv) [inline C/cp/sum of triplet.c:87]: @akanksha-saini add comments after this `/*`. Also, add space complexity.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

**Review by iamrajiv — APPROVED** [review]
@akanksha-saini LGTM! Nice work.

File: C/cp/sum of triplet.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
+#include<stdio.h>
+
+void sort(int a[1000], int n)
```
Comment (siddharth25pandey) [inline C/cp/sum of triplet.c]: Take dynamic size array

File: C/cp/sum of triplet.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+			{
+				temp = a[j];
+				a[j] = a[j + 1];
+				a[j + 1] = temp;
+			}
+		}
+	}
+}
+
+void tripletSum(int a[1000], int n, int target)
```
Comment (siddharth25pandey) [inline C/cp/sum of triplet.c]: Do the same, just write **int a[]**

File: C/cp/sum of triplet.c [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+			{
+				j++;
+			}
+		}
+	}
+}
+
+int main()
+{
+	int a[1000], n, i, target;
```
Comment (siddharth25pandey) [inline C/cp/sum of triplet.c]: Declare the array size after taking input from the user.
i.e **int a[n]**

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@akanksha-saini Do the required changes

**Comment by akanksha-saini** [comment]
Please review now @siddharth25pandey , I have made the requested changes.

File: C/cp/sum of triplet.c [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+}
+
+int main()
+{
+<<<<<<< HEAD
+	int *a, n,i,target;
+	
+	scanf("%d",&n); //input no. of elements in array
+	
+	a = (int)malloc(n * sizeof(int)); // take dynamic array as they are not of fixed size and are allocated on heap
```
Comment (siddharth25pandey) [inline C/cp/sum of triplet.c]: here, do the changes,
a=(int*)malloc

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

**Comment by siddharth25pandey** [comment]
@akanksha-saini Without using **stdlib** header how you are using malloc & allocating the memory in heap??
First run your Program, than try to submit the PR for review.
You can do either of two
*  **int a[n]** by defining n above this line **OR**
* **use Dynamic Memory allocation** by including standard Library

**Comment by akanksha-saini** [comment]
I have committed changes to the file @siddharth25pandey

**Review by siddharth25pandey — APPROVED** [review]
@akanksha-saini Looks fine to me
Able to merge
Thanks!

**Comment by akanksha-saini** [comment]
Please review my pull request @SKAUL05

**Review by SKAUL05 — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @akanksha-saini for code

**Pull request merged**